account-kit/signer/src/smartAccountSigner.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  hashMessage,
+  hashTypedData,
+  type Address,
+  type Hex,
+  type SignableMessage,
+  type TypedData,
+  type TypedDataDefinition,
+} from "viem";
+import type { BaseSignerClient } from ".";
+import {
+  unpackSignRawMessageBytes,
+  type SmartAccountSigner,
+} from "@aa-sdk/core";
+import { hashAuthorization, type Authorization } from "viem/experimental";
+
+/**
+ * SmartAccountSignerFromClient is a class that implements the SmartAccountSigner interface.
+ * It wraps a BaseSignerClient and provides methods for signing messages, typed data, and authorizations.
+ */
+export class SmartAccountSignerFromClient implements SmartAccountSigner {
```

Comment:
naming is hard

Reply:
`BaseAlchemySigner` already implements `SmartAccountAuthenticator` which extends `SmartAccountSigner`

i need to see how this is used (still reviewing), but just a heads up in case that means anything here